### PR TITLE
Closes #724 - improve documentation

### LINKED ISSUE #724: Improve documentation of `derive_vars_query()`
Testers feedback:
> Not quite sure what this function is for.

I would propose
- to add some context on the function (for which datasets and CDISC variables it is intended for) and
- to add a link to the "Queries Dataset Documentation" vignette.

### PR DESCRIPTION
**Pull request by yangc71**
Closes #724.

Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/) 
- [x] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiral/articles/unit_test_guidance.html#writing-unit-tests-in-admiral-)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiral/articles/programming_strategy.html#deprecation-1)?
- [x] Update to all relevant roxygen headers and examples 
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by yangc71** [comment]
Hi @bms63, I just realized Stefan is out this week, if you got some time can you please help review? Also the lint check is failing but it's running fine for me locally, can you help provide suggestions on how to fix? Thanks much.

**Comment by bms63** [comment]
When I run `lintr::lint_package()` I get these two issues with the urls.  

image

File: R/derive_vars_query.R [hunk 1]
```diff
@@ -1,6 +1,19 @@
 #' Derive Query Variables
 #'
-#' @details For each unique element in `VAR_PREFIX`, the corresponding "NAM"
+#' @details This function can be used to derive CDISC variables such as
+#'   `SMQzzNAM`, `SMQzzCD`, `SMQzzSC`, `SMQzzSCN`, and `CQzzNAM` in ADAE and
+#'   ADMH, and variables such as `SDGzzNAM`, `SDGzzCD`, and `SDGzzSC` in ADCM.
+#'   An example usage of this function can be found in the
+#'   [OCCDS vignette](https://r.roche.com/s/71c86bafe62a4200fced1/files
```
Comment (bms63) [inline R/derive_vars_query.R:7]: Are these links valid?  Why do they have roche in them?

Comment (yangc71) [reply R/derive_vars_query.R]: You are right...updated to pharmaverse, do the links work now? Also there is a R CMD warning "Missing link or links in documentation object" about these objects even though the link is specified, I'm not sure which part is missing? Thanks for your help.

**Comment by bms63** [comment]
@yangc71 I think the internal links need to be written like this 
image

**Comment by yangc71** [comment]
Thanks @bms63, need to learn more about links...hope they all work this time?

**Comment by bms63** [comment]
> Thanks @bms63, need to learn more about links...hope they all work this time?

More suffering than learning lol.

**Review by bundfussr — APPROVED** [review]

**Pull request merged**